### PR TITLE
RAC: `ListBox`

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

File: packages/components/src/ListBox/_ListBoxSection.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import { useListBoxContext } from './Context';
+
+export interface SectionProps extends RAC.SectionProps<object> {}
+
+const _Section = (props: SectionProps) => {
+  const { classNames } = useListBoxContext();
+  return (
+    <Section
+      className={cn(classNames.section, classNames.sectionTitle)}
```
[the remark this answers is not in the document]
Comment (OsamaAbdellateef) [reply packages/components/src/ListBox/_ListBoxSection.tsx:14]: I could move styles from `sectionTitle` and add them to the `section` styles but i think having `sectionTitle` alone will make it resuable for any future chaneges 
what do you think ? 🤔 @sebald

[the remark this answers is not in the document]
Comment (OsamaAbdellateef) [reply packages/components/src/ListBox/_ListBoxSection.tsx:14]: Do u want me to merge them ? @sebald

**Comment by sarahgm** [comment]
@OsamaAbdellateef  there are uncovered lines

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef there are uncovered lines

Yeah because it doesn't have tests for it 
but when it is used with combobox & select component it will  be covered hopefully 
and if it doesn't i create tests for it

File: themes/theme-core/src/components/ListBox.styles.ts [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-    // Need to use data attributes here because `focus-visible` only works with <Select>
-    'data-[focus-visible]:text-text-light data-[focus-visible]:bg-highlight',
-    'aria-enabled:hover:text-text-light aria-enabled:hover:bg-highlight',
-    'aria-selected:text-text-light aria-selected:bg-highlight',
-    'aria-disabled:text-text-disabled aria-disabled:cursor-not-allowed',
+    font(),
+    'cursor-pointer px-1.5 py-0.5 outline-none',
+    'data-[focused]:text-text-light data-[focused]:bg-highlight',
+    'data-[selected]:text-text-light aria-selected:bg-highlight',
+    'data-[disabled]:text-text-disabled aria-disabled:cursor-not-allowed',
```
Comment (sarahgm) [inline themes/theme-core/src/components/ListBox.styles.ts:15]: enabled:hover styles are missing or am I wrong?

Comment (sarahgm) [reply themes/theme-core/src/components/ListBox.styles.ts:15]: @OsamaAbdellateef  don't know if you saw this already :)

Comment (OsamaAbdellateef) [reply themes/theme-core/src/components/ListBox.styles.ts:15]: i am not getting this @sarahgm

**Review by sarahgm — APPROVED** [review]

**Pull request merged**